src/input_handler.cc

```diff
@@ -810,6 +810,7 @@ class Prompt : public InputMode
         else if (key == Key::Escape or key == ctrl('c'))
         {
             history_push(line);
+quit:
```

Comment:
The behavior seems reasonable. (The implementation could be done by simply extending the condition above since `history_push("")` is a nop.)